src/machine.rs

```diff
@@ -1572,3 +1578,19 @@ pub(crate) async fn promise_result_to_future(
         }
     }
 }
+
+async fn stream_to_json_array<T, S>(mut stream: Pin<&mut S>) -> Result<String, anyhow::Error>
+where
+    T: Serialize,
+    S: Stream<Item = T>,
+{
+    let mut stream_json = vec![];
+    let mut ser = serde_json::Serializer::new(&mut stream_json);
+    let mut seq = ser.serialize_seq(None).map_err(|e| anyhow::Error::msg(e.to_string()))?;
+    while let Some(key) = stream.next().await {
+        seq.serialize_element(&key).map_err(|e| anyhow::Error::msg(e.to_string()))?;
```

Comment:
nit: for both, can `anyhow` not figure out how to wrap the error?

```suggestion
    let mut seq = ser.serialize_seq(None)?;
    while let Some(key) = stream.next().await {
        seq.serialize_element(&key)?;
```

Reply:
Brilliant, thank you - done